### DISCUSSION TITLE
Add https redirect to port 80

### DISCUSSION THREAD

**Discussion by Eeemil** [nextcloud/all-in-one#667] [Ideas] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. install nextcloud from https://marketplace.digitalocean.com/apps/nextcloud
2. follow installation steps, wait for containers to start
3. curl `server.name.tld`, get "connection refused on port 80"

### Expected behavior 
A HTTPS redirect
### Actual behavior 
Connection refused

### Host OS 
Ubuntu 20.04 via https://marketplace.digitalocean.com/apps/nextcloud

#### Nextcloud AIO version 
v1.2.0
#### Current channel 
latest
#### Other valuable info 

I "resolved" the issue by `docker exec -it nextcloud-aio-mastercontainer` and added the following:
```
Listen 80

<VirtualHost *:80>
    RewriteEngine On
    RewriteCond %{HTTPS} off
    RewriteRule ^ https://%{HTTP_HOST}%{REQUEST_URI}
</VirtualHost>
```
to the top of /etc/apache2/sites-enabled/mastercontainer.conf and this resolved the issue. However, I'm afraid this will either be overwritten during updates or that the letsencrypt certificate will be unable to update.

**Comment by szaimen** [comment]
Hello, may I ask why you need this redirect? Does any app or program not work without this?

**Reply by Eeemil** [reply]
When providing a link to the nextcloud instance, such as `nextcloud.example.com`, some users will just write `nextcloud.example.com` which defaults to HTTP (port 80) in some browsers and get "connection refused".

**Reply by szaimen** [reply]
I see. Can you try to edit the `/Caddyfile` in the mastercontainer and remove this line?
https://github.com/nextcloud/all-in-one/blob/159a46db524e08e93e797e5bd9b4492e8c29964e/Containers/mastercontainer/Caddyfile#L2
Then please restart the container.

Does this work?

**Reply by Eeemil** [reply]
Almost! It provides a HTTPS redirect to `https://sitename:8443/`. I digged a bit deeper in caddy, and the auto_https probably assumes it should provide a redirect for the `https://:8443` service.

It works perfectly with this configuration though:
```
{
    auto_https disable_redirects
    storage file_system {
        root /mnt/docker-aio-config/caddy/
    }
}

http://:80 {
  redir https://{host}{uri}
}

https://:8443 {

    reverse_proxy localhost:8000

    tls {
        on_demand
    }
}
```

**Reply by szaimen** [reply]
Great, perfect! Can you create a pull request with this improvement? :)

**Reply by Eeemil** [reply]
Sure thing, https://github.com/nextcloud/all-in-one/pull/670

**Reply by szaimen** [reply]
Thank you very much! :)

**Comment by szaimen** [comment]
This will be implemented with https://github.com/nextcloud/all-in-one/pull/670

**Comment by szaimen** [comment, marked as answer]
Hello, this is now released with 1.3.0 Beta. Feel free to try it out and report back! Thank you! :)
See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
